code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -827,6 +846,51 @@
 	succeed_activate()
 	add_cooldown()
 
+// ***************************************
+// *********** Spawn hivemind
+// ***************************************
+/datum/action/xeno_action/spawn_hivemind
+	name = "Create hivemind"
+	action_icon_state = "lay_hivemind"
+	plasma_cost = 400
+	cooldown_timer = 300 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LAY_HIVEMIND
+
+
+/datum/action/xeno_action/spawn_hivemind/action_activate()
+	var/mob/living/carbon/xenomorph/queen/xeno = owner
+	var/turf/current_turf = get_turf(owner)
+
+	var/obj/effect/alien/weeds/alien_weeds = locate() in current_turf
+	if(!alien_weeds)
+		to_chat(owner, "<span class='warning'>We can't place a hivemind here. Lay it on some resin.</span>")
+		return FALSE
+
+	if(length(xeno.hive.xenos_by_typepath[/mob/living/carbon/xenomorph/hivemind]))
+		to_chat(owner, "<span class='warning'>We already have a hivemind active.</span>")
+		return FALSE
+
+	if(!do_after(owner, 3 SECONDS, FALSE, alien_weeds))
+		return FALSE
+
+	if(!current_turf.check_alien_construction(owner))
+		return FALSE
+
+	owner.visible_message("<span class='xenowarning'>\The [owner] has created a hivemind!</span>", \
+		"<span class='xenowarning'>We have created a hivemind!</span>")
+
+	var/obj/effect/alien/weeds/node/hivemindcore/core = new /obj/effect/alien/weeds/node/hivemindcore(current_turf)
+	if(!isxenohivemind(core.parent))
+		return FALSE
+	var/mob/living/carbon/xenomorph/hivemind/xeno_hivemind = core.parent
+	playsound(xeno_hivemind.loc, 'sound/effects/splat.ogg', 25)
+
+	xeno_hivemind.offer_mob()
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_OPEN_TIMED_SHUTTERS_XENO_HIVEMIND)
```

Comment:
Why does the creation of the hivemind open the shutters?

Reply:
because the balance issue that people were complaining you could place a hivemind behind the fog / shutters. They wanted to be able to take it out, so we agreed to open shutters / remove fog.

Reply:
They shouldn't be able to put a hivemind behind shutters, as xenos can't access that area, right?

On fog, that's a bit more problematic. How are you handling it?
First solution that comes to my mind is, if the game mode is one with permafog, runnig a locate fog in range when checking if the hivemind can be placed.

Reply:
I was just removing the fog.
Xenos are opting for a hivemind with the downside of no fog.
Saved doing anything weird /  complicated for it.

Reply:
This signal is raising the shutters, not removing the fog.